spec/text-editor-component-spec.js

```diff
@@ -1896,6 +1896,9 @@ describe('TextEditorComponent', () => {
       const decoration = editor.decorateMarker(marker, {type: 'overlay', item: overlayElement, class: 'a'})
       await component.getNextUpdatePromise()
 
+      let overlayComponent
+      component.overlayComponents.forEach(c => overlayComponent = c)
```

Comment:
It looks like you are trying to assign the last element of array to the `overlayComponent ` variable. Perhaps it could be grabbed from the array directly?

Reply:
Yea, sorry for the odd syntax. `overlayComponents` is a `Set`. In `Set`, items don't have positions, but you can iterate over all the items in the `Set`. In the case of this test, I know there's only going to be one item.

Reply:
Very minor, but this might be slightly clearer:

```js
const overlayComponent = component.overlayComponents.values().next().value
```

Reply:
Ah, yea. That's better. I hadn't looked at the `Iterator` api. Will Change.

Reply:
Why do you know there is only one item?